cli/azd/pkg/containerapps/container_app.go

```diff
@@ -118,44 +134,49 @@ func (cas *containerAppService) persistSettings(
 	subscriptionId string,
 	resourceGroupName string,
 	appName string,
-	obj map[string]any) (map[string]any, error) {
+	obj map[string]any,
+	options *ContainerAppOptions,
+) (map[string]any, error) {
 	shouldPersistDomains := cas.alphaFeatureManager.IsEnabled(persistCustomDomainsFeature)
 	shouldPersistIngressSessionAffinity := cas.alphaFeatureManager.IsEnabled(persistIngressSessionAffinity)
 
 	if !shouldPersistDomains && !shouldPersistIngressSessionAffinity {
 		return obj, nil
 	}
 
-	aca, err := cas.getContainerApp(ctx, subscriptionId, resourceGroupName, appName)
+	aca, err := cas.getContainerApp(ctx, subscriptionId, resourceGroupName, appName, options)
 	if err != nil {
 		log.Printf("failed getting current aca settings: %v. No settings will be persisted.", err)
 	}
 
-	if aca == nil ||
-		aca.Properties == nil ||
-		aca.Properties.Configuration == nil ||
-		aca.Properties.Configuration.Ingress == nil {
-		// no settings to persist
-		return obj, nil
+	var ingress *armappcontainers.Ingress
```

Comment:
Converting to a model type here means we could drop fields if a newer API version introduced new fields, I think?  Is it safe to do this?

Reply:
We can make this loosely typed everywhere but wanted to get some feedback before making all these changes.